Core/include/Acts/Seeding/SeedFinderUtils.ipp

```diff
@@ -50,14 +50,73 @@ LinCircle transformCoordinates(external_spacepoint_t& sp,
   l.cotTheta = cot_theta;
   l.Zo = zM - rM * cot_theta;
   l.iDeltaR = iDeltaR;
-  l.U = x * iDeltaR2;
-  l.V = y * iDeltaR2;
+  l.U = xNewFrame * iDeltaR2;
+  l.V = yNewFrame * iDeltaR2;
   l.Er = ((varianceZM + varianceZSP) +
           (cot_theta * cot_theta) * (varianceRM + varianceRSP)) *
          iDeltaR2;
   return l;
 }
 
+template <typename external_spacepoint_t>
+LinCircle transformCoordinates(
+    InternalSpacePoint<external_spacepoint_t>& sp,
+    const InternalSpacePoint<external_spacepoint_t>& spM,
+    const std::array<float, 6> transformVariables) {
+  auto extractFunction =
+      [](const InternalSpacePoint<external_spacepoint_t>& obj)
+      -> std::array<float, 6> {
+    std::array<float, 6> output{obj.x(),      obj.y(),         obj.z(),
+                                obj.radius(), obj.varianceR(), obj.varianceZ()};
+    return output;
+  };
+
+  return transformCoordinates<InternalSpacePoint<external_spacepoint_t>>(
+      sp, spM, extractFunction, transformVariables);
+}
+
+template <typename external_spacepoint_t, typename callable_t>
+LinCircle transformCoordinates(external_spacepoint_t& sp,
+                               const external_spacepoint_t& spM,
+                               callable_t&& extractFunction,
+                               const std::array<float, 6> transformVariables) {
+  // The computation inside this function is exactly identical to that in the
+  // vectorized version of this function, except that it operates on a single
+  // spacepoint. Please see the other version of this function for more
+  // detailed comments.
+
+  auto [xM, yM, zM, rM, varianceRM, varianceZM] = extractFunction(spM);
```

Comment:
I suppose this is the main issue... for every time we call this function we allocate this memory and then de-allocate. right?

Reply:
Thank you for your comments. I agree this is a problem. I could remove extractFunction entirely and only call for the variances which are the only values that we need, do you think this would be better?

Reply:
yes, I agree, but these function calls are no different from what happens in Athena, so I am not sure if that explains the time difference